doc/reference/guidelines.rst

```diff
@@ -0,0 +1,48 @@
+.. highlight:: python
+
+=================
+Coding Guidelines
+=================
+
+Python
+======
+
+- PEP8 ignore E501
+- prefer % over format
+- try to avoid generators and decorators
```

Comment:
Try to LEARN using generators you mean. List comprehension, dict comprehension, and basic manipulation using map, filter, sum, ... Of course those are not always useful, but in many cases it makes the code easier to read. But if the code is easier to read in 5 lines, write 5 lines. Readability > Python-guru-master.